docs/4_3_co-simulation_schema.adoc

```diff
@@ -73,6 +73,189 @@ unsignedInt: 0 +
+
+|`compiler`
+|Compiler, e.g. gcc, VisualStudio
+
+|`compilerVersion`
+|Compiler version, e.g. 3.2
+
+|====
+
+==== Definition of preprocessor definitions (PreProcessorDefinitions)
+
+The element `PreProcessorDefinitons` is optionally present. It is only meaningful for source code FMUs.
+It contains a list of compiler macros required to compile the sources. All defines must be set.
+
+Element "*PreProcessorDefinitions*" of `CoSimulation` is defined as:
+
+image::images/PreProcessorDefinitions_schema.png[]
+
+Element `PreProcessorDefinitions` contains 1..n elements `Define`
+The attributes of element "*Define*" have the following meaning (all attributes are optional with exception of `name`):
+
+[cols="1,1",options="header"]
+|====
+|Attribute Name
+|_Description_
+
+|`name`
+|Case sensitive macro name e.g. LOGLEVEL
+
+|`value`
+|Optional value of the macro, e.g. 3 or ON
+
+|`description`
+|Optional description what the behaviour is controlled by the macro.
+
+|`platform`
+|Platform name, e.g. x86_64-windows, i686-linux,...
+platform attribute must only be added if source file is platform-dependent.
+
+|`compiler`
+|Compiler, e.g. gcc, VisualStudio
+
+|`compilerVersion`
+|Compiler version, e.g. 3.2
+
+|====
+
+==== Definition of libraries to be linked (Libraries)
+
+The element `Libraries` is optionally present. It contains an ordered list of libraries (element `File`) to be presented to the linker. Link order: First one wins.
+
+Element "*Libraries*" of `CoSimulation` is defined as:
+
+image::images/Libraries_schema.png[]
+
+Element `Libraries` contains 1..n elements `File`
+The attributes of element "*File*" have the following meaning (all attributes are optional with exception of `name`):
+
+[cols="1,1",options="header"]
+|====
+|Attribute Name
+|_Description_
+
+|`name`
+|Name of the library to be linked with, to be found in the platform/compiler directories of the FMU.
+
+|`platform`
+|Platform name, e.g. x86_64-windows, i686-linux,...
+platform attribute must only be added if source file is platform-dependent.
+
+|`compiler`
+|Compiler, e.g. gcc, VisualStudio
+
+|`compilerVersion`
+|Compiler version, e.g. 3.2
+
+|====
+
+==== Definition of External Dependencies (ExternalDependencies)
+
+The element `ExternalDependencies` is optionally present. It contains a list of external dependencies (element `ExternalDependency`), e.g. external libraries not contained in the FMU.
+
+Element "*ExternalDependencies*" of `CoSimulation` is defined as:
+
+image::images/ExternalDependencies_schema.png[]
+
+Element `ExternalDependencies` contains 1..n elements `ExternalDependency`
+The attributes of element "*ExternalDependency*" have the following meaning (all attributes are optional with exception of `name`):
+
+[cols="1,1",options="header"]
+|====
+|Attribute Name
+|_Description_
+
+|`name`
+|Name of the library to be linked with, to be found in the platform/compiler directories of the FMU.
```

Comment:
I assume this a misstake, if it's an external dependency is shouldn't be found in the FMU or?

Reply:
Yes, it is a mistake. I corrected it.